src/Core/Framework/App/Subscriber/SystemLanguageChangedSubscriber.php

```diff
@@ -0,0 +1,113 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Framework\App\Subscriber;
+
+use Shopware\Administration\Snippet\AppAdministrationSnippetCollection;
+use Shopware\Administration\Snippet\AppAdministrationSnippetEntity;
+use Shopware\Core\Defaults;
+use Shopware\Core\Framework\App\AppCollection;
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepository;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Maintenance\System\Service\SystemLanguageChangeEvent;
+use Shopware\Core\System\Language\LanguageCollection;
+use Shopware\Core\System\Language\LanguageEntity;
+use Shopware\Core\System\Locale\LocaleEntity;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * @internal
+ */
+#[Package('framework')]
+readonly class SystemLanguageChangedSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @param EntityRepository<LanguageCollection> $languageRepository
+     * @param EntityRepository<AppCollection> $appRepository
+     * @param EntityRepository<AppAdministrationSnippetCollection>|null $snippetRepository
+     */
+    public function __construct(
+        private EntityRepository $languageRepository,
+        private EntityRepository $appRepository,
+        private ?EntityRepository $snippetRepository,
+    ) {
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            SystemLanguageChangeEvent::class => 'onSystemLanguageChanged',
+        ];
+    }
+
+    public function onSystemLanguageChanged(SystemLanguageChangeEvent $event): void
+    {
+        // This repository is part of the Administration bundle and not necessarily installed and available.
+        if ($this->snippetRepository === null) {
+            return;
+        }
+
+        $context = Context::createDefaultContext();
+
+        $snippets = $this->getSnippets($this->snippetRepository, $context);
+        if ($snippets->count() === 0) {
+            return;
+        }
+
+        $appIds = $this->getApps($context)->getIds();
+
+        $appsWithSnippets = [];
+        foreach ($snippets as $snippet) {
+            if (\in_array($snippet->getAppId(), $appIds, true) && !\in_array($snippet->getAppId(), $appsWithSnippets, true)) {
```

Comment:
The test for `\in_array($snippet->getAppId(), $appIds, true)` makes no sense to me. Are you checking if a snippet's `app_id` exists in the `app` table? This is secured by data base constraints. :D

```suggestion
        $appsWithSnippets = [];
        foreach ($snippets as $snippet) {
            $appsWithSnippets[] = $snippet->getAppId();
       }
       
       $appsWithSnippets = array_unique($appsWithSnippets);
```

Reply:
Yeah IDK where my brain went yesterday evening 🤣